### PR TITLE
HTML_Validation manually checked

### LINKED ISSUE #2798: Fix HTML Validator errors for templates
Child issue of #163. Aims to resolve all HTML validation issues associated with templates.

### Related files
Folders
- [ ] account
- [ ] authors
- [ ] books
- [ ] contact
- [ ] covers
- [ ] email
- [ ] history
- [ ] home
- [ ] languages
- [ ] lib
- [ ] libraries
- [ ] lists
- [ ] merge	
- [ ] publishers
- [ ] recentchanges
- [ ] search
- [ ] site	
- [ ] stats	
- [ ] subjects	
- [ ] type	
- [ ] work

Files
- [x] account.html
- [x] borrow_admin.html 
- [ ] borrow_admin_no_update.html [IN PROGRESS]
- [x] design.html	
- [ ] diff.html	
- [ ] edit_yaml.html	
- [ ] editpage.html	
- [ ] form.html	
- [ ] history.html	
- [ ] internalerror.html	
- [ ] login.html
- [ ] message.html
- [ ] messages.html
- [ ] notfound.html	
- [ ] permission.html	
- [ ] permission_denied.html	
- [ ] showamazon.html	
- [ ] showia.html	
- [ ] showmarc.html	
- [ ] site.html	
- [ ] status.html	
- [x] subjects.html
- [ ] support.html	
- [ ] viewpage.html	
- [ ] widget.html	
- [ ] work_search.html	

### Stakeholders
@jdlrobson @mekarpeles @koderjoker

### PR DESCRIPTION
**Pull request by ArunTeltia**
Partially resolve #2798

there is a problem with borrow_admin.html there are two id with some name 

can I change this to class and then I will change all the files that contain the id?

### Stakeholders
@jdlrobson @mekarpeles

### CODE REVIEW

**Comment by ArunTeltia** [comment]
before changing attribute id to class of borrowTable

after changing

**Comment by ArunTeltia** [comment]
@jdlrobson can you check this

File: openlibrary/templates/account/create.html [hunk 1]
```diff
@@ -63,7 +63,7 @@ <h1>$_("Sign Up")</h1>
                 </div>
             </div>
 
-        <small>$:_("By signing up, you agree to the Internet Archive's <a href='//archive.org/about/terms.php' target='_blank'>Terms of Service</a>.")</small>
+        <small>$:_("By signing up, you agree to the Internet Archive's <a href="//archive.org/about/terms.php" target="_blank">Terms of Service</a>.")</small>
```
Comment (cclauss) [inline openlibrary/templates/account/create.html:66]: CAUTION!  The outer string is double-quoted and then inner strings are single-quoted.  Making all strings double-quoted is changing the meaning of the outer string.

File: openlibrary/templates/account/password/reset_success.html [hunk 2]
```diff
@@ -7,5 +7,5 @@ <h1>$_("Hi, %(user)s", user=username)!</h1>
 </div>
 
 <div id="contentBody">
-    <p>$:_("Your password has been updated. Please <a href='/account/login?redirect=/'>log in to continue</a>.")</p>
+    <p>$:_("Your password has been updated. Please <a href="/account/login?redirect=/">log in to continue</a>.")</p>
```
Comment (cclauss) [inline openlibrary/templates/account/password/reset_success.html:10]: CAUTION: Outer string vs. inner strings.

File: openlibrary/templates/account/verify/success.html [hunk 3]
```diff
@@ -8,5 +8,5 @@ <h1>$_("Hi, %(user)s", user=account.displayname)!</h1>
 
 <div id="contentBody">
   $ link = "/account/login?redirect=/&username=" + url_quote(account.email)
-  <p>$:_("Yay! Your email address has been verified. Please <a href='%s'>log in to continue</a>.", link)</p>
+  <p>$:_("Yay! Your email address has been verified. Please <a href="%s">log in to continue</a>.", link)</p>
```
Comment (cclauss) [inline openlibrary/templates/account/verify/success.html:11]: CAUTION: Outer string vs. inner strings.

File: openlibrary/templates/admin/inspect/thing.html [hunk 4]
```diff
@@ -1,30 +1,31 @@
...
-    <h1>Inspect thing</h1>
+  $:render_template("admin/menu")
+  <h1>Inspect thing</h1>
 </div>
 
 <div id="contentBody">
   <h2>Offline tasks associated with this item</h2>
   $:render_template("admin/inspect/tasks", tasks)
-</div>
+</div>
```
Comment (cclauss) [inline openlibrary/templates/admin/inspect/thing.html:31]: Please add just one carriage return to the end of the file so the GitHub code viewer removes the ⛔.

File: openlibrary/templates/books/check.html [hunk 5]
```diff
@@ -53,7 +53,7 @@ <h3 class="booktitle"><a href="javascript:;" title="$_('Select this book')">$wor
 </div>
 
 <script type="text/javascript">
-<!--
+
```
Comment (cclauss) [inline openlibrary/templates/books/check.html:56]: Was this code commented out for a reason?  Was the commenting removed for a reason?

File: openlibrary/templates/books/edit/edition.html [hunk 6]
```diff
@@ -432,7 +432,7 @@
                 <span class="tip">$_("The name of the publisher's series.") $_("For example:") <i>The Story of Civilization, Part III</i></span>
             </div>
             <div class="input">
-                <input type="text" name="edition--series--0" id="edition-series" value="$(book.series and book.series[0] or "")"/>
+                <input type="text" name="edition--series--0" id="edition-series" value="$(book.series and book.series[0] or )"/>
```
Comment (cclauss) [inline openlibrary/templates/books/edit/edition.html:435]:
```suggestion
                <input type="text" name="edition--series--0" id="edition-series" value="$(book.series and book.series[0] or '')"/>
```

File: openlibrary/templates/work_search.html [hunk 7]
```diff
@@ -43,7 +43,7 @@
 $ start_facet_count = 5
 $ facet_inc = 10
 <script type="text/javascript">
-<!--
```
Comment (cclauss) [inline openlibrary/templates/work_search.html:46]: Is the commenting removed for a reason?  Was there a commenting close that also needs to be deleted?

File: openlibrary/templates/type/home/edit.html [hunk 8]
```diff
@@ -6,36 +6,34 @@
...
             $:macros.EditButtons()
-        </form>
-        $:macros.HelpFormatting()
+            $:macros.HelpFormatting()
+        </div>
     </div>
-</div>
-
-</div>
+</form>
```
Comment (cclauss) [inline openlibrary/templates/type/home/edit.html:39]: ⛔

File: openlibrary/templates/type/edition/view.html [hunk 9]
```diff
@@ -79,9 +79,9 @@
         schema_attribute = cond(include_schema, ' itemprop="author"', '')
         authors = cond(work, work and work.get_authors(), edition and edition.get_authors())
         if authors:
-            return ', '.join('<a href="%s"%s>%s</a>' % (a.url(), schema_attribute, truncate(a.name, 40)) for a in authors)
+            return ', '.join('<a href="%s">%s</a>' % (a.url(), schema_attribute, truncate(a.name, 40)) for a in authors)
```
Comment (cclauss) [inline openlibrary/templates/type/edition/view.html:82]: There are three variables `a.url(), schema_attribute, truncate(a.name, 40)` so their should also be three `%s`.

File: openlibrary/templates/type/edition/view.html [hunk 10]
```diff
@@ -79,9 +79,9 @@
...
         elif edition.author_names:
-            return ('<span%s>' % (schema_attribute)) + ", ".join(edition.author_names) + "</span>"
+            return ('<span>' % (schema_attribute)) + ", ".join(edition.author_names) + "</span>"
```
Comment (cclauss) [inline openlibrary/templates/type/edition/view.html:84]: There is one substitution variable `schema_attribute` so there should also be one `%s`.

**Review by cclauss — CHANGES_REQUESTED** [review]
There is a lot of good stuff here but there are also things that won't work.  Reviewing 33 files in a single PR is difficult for maintainers unless the changes are fully automatic.  I would suggest multiple PRs that touch fewer files.

File: openlibrary/templates/type/edition/view.html (hunk 10 above)
Comment (ArunTeltia) [reply openlibrary/templates/type/edition/view.html:84]: Okay I will change that and commit in multiple PR but mostly the commits are of indentation so I thought it will be easy for being reviewed 
Thank you for telling me that I will do a multiple PR of this

**Comment by ArunTeltia** [comment]
Thank you for telling me all these changes ,I will do all the change and do a PR

File: openlibrary/templates/books/check.html (hunk 5 above)
Comment (ArunTeltia) [reply openlibrary/templates/books/check.html:56]: It is not the comment in the script the comment is js is written like 
```/* */``` not ```<--``` 
It was giving an error thats why I remove this line 
This type of comment is used in HTML not in script
Screenshot (29)

File: openlibrary/templates/work_search.html (hunk 7 above)
Comment (ArunTeltia) [reply openlibrary/templates/work_search.html:46]: https://github.com/internetarchive/openlibrary/pull/3977#discussion_r515615618
same issue

File: openlibrary/templates/account/create.html (hunk 1 above)
Comment (ArunTeltia) [reply openlibrary/templates/account/create.html]: Fixed this is the latest PR

File: openlibrary/templates/account/password/reset_success.html (hunk 2 above)
Comment (ArunTeltia) [reply openlibrary/templates/account/password/reset_success.html]: Fixed this  in the latest PR

File: openlibrary/templates/account/verify/success.html (hunk 3 above)
Comment (ArunTeltia) [reply openlibrary/templates/account/verify/success.html]: Fixed this in the latest PR

File: openlibrary/templates/admin/inspect/thing.html (hunk 4 above)
Comment (ArunTeltia) [reply openlibrary/templates/admin/inspect/thing.html]: This is also resolved

File: openlibrary/templates/books/edit/edition.html (hunk 6 above)
Comment (ArunTeltia) [reply openlibrary/templates/books/edit/edition.html]: This is resolved as well

**Comment by ArunTeltia** [comment]
I had resolved all the issue that you mentioned in this PR @cclauss Thank you for helping me

File: openlibrary/macros/FulltextSnippet.html [hunk 11]
```diff
@@ -12,7 +12,7 @@
       $for snippet in snippets:
         $if snippet: 
           <li class="fulltext-excerpt">
-            &hellip;$:(snippet.replace("<", "&laquo;").replace(">", "&raquo;").replace("{{{", "<mark class='highlight'><strong>").replace("}}}", "</strong></mark>"))&hellip;
+            &hellip;$:(snippet.replace("<", "&laquo;").replace(">", "&raquo;").replace("{{{", "<mark class="highlight"><strong>").replace("}}}", "</strong></mark>"))&hellip;
```
Comment (jdlrobson) [inline openlibrary/macros/FulltextSnippet.html:15]: This doesn't seem related to HTML validation. It changes a single quote to a double quote, but either is valid.
Did you intend to add this?

File: openlibrary/macros/RecentPageList.html [hunk 12]
```diff
@@ -5,19 +5,19 @@
 $ pages = list_recent_pages(path, limit=limit, offset=page * limit)
 
 <table>
-$for p in pages:
+    $for p in pages:
```
Comment (jdlrobson) [inline openlibrary/macros/RecentPageList.html:8]: These changes seem unrelated to the change being made. Indentation shouldn't have any impact on HTML validation.

I'd suggest creating a patch on master e.g. `git diff origin/master HTML_validation > patch.diff` and then carefully applying it on a fresh branch removing unrelated changes like these.

File: openlibrary/macros/YouTube.html [hunk 13]
```diff
@@ -6,8 +6,8 @@
 
 $if id in allowed_videos:
     <object width="640" height="385">
-        <param name="movie" value="http://www.youtube.com/v/$id&hl=en_US&fs=1&"></param>
-        <param name="allowFullScreen" value="true"></param>
-        <param name="allowscriptaccess" value="always"></param>
-        <embed src="http://www.youtube.com/v/$id&hl=en_US&fs=1&" type="application/x-shockwave-flash" allowscriptaccess="always" allowfullscreen="true" width="640" height="385"></embed>
+        <param name="movie" value="http://www.youtube.com/v/$id&hl=en_US&fs=1&" />
```
Comment (jdlrobson) [inline openlibrary/macros/YouTube.html:9]: The previous setup was valid HTML. Why is this better? Was this an intended change?

**Review by jdlrobson — CHANGES_REQUESTED** [review]
There's a lot of good stuff in here, but also a lot of unrelated changes such as changing single quotes to double-quotes.

Keep the patch as simple as possible. As a volunteer reviewer it makes reviewing much easier if the patches are tightly scoped to the issue at hand.

File: openlibrary/macros/RecentPageList.html (hunk 12 above)
Comment (ArunTeltia) [reply openlibrary/macros/RecentPageList.html:8]: Thank you for telling me this, I will do this, and for now, I will recheck all the things that I changed, I accidentally did some single commas to double commas, and I will try to resolve these

File: openlibrary/macros/FulltextSnippet.html (hunk 11 above)
Comment (ArunTeltia) [reply openlibrary/macros/FulltextSnippet.html:15]: I read somewhere that " " for links are better than ' ' in HTML, that's why I change those otherwise there was no problem whatsoever

File: openlibrary/macros/YouTube.html (hunk 13 above)
Comment (ArunTeltia) [reply openlibrary/macros/YouTube.html:9]: param and embed tags give an warning when they are closed 
https://www.w3schools.com/tags/tag_param.asp
that's why I used this

File: openlibrary/templates/admin/loans.html [hunk 14]
```diff
@@ -22,7 +22,7 @@ <h1>Outstanding eBook Loans</h1>
                 <input type="submit" value="Update ACS4 loan statuses" />
             </form>
         $else:
-            <span class="adminOnly" style="font-size: 24px; line-height: 1.5em;"><i>THIS PAGE ISN'T UPDATING LIVE.</u><br />We're currently collecting stats - bulk status update is disabled.  <br />Please do not refresh loan status manually, thanks.</span>
+            <span class="adminOnly" style="font-size: 24px; line-height: 1.5em;"><u>THIS PAGE ISN'T UPDATING LIVE.</u><br />We're currently collecting stats - bulk status update is disabled.  <br />Please do not refresh loan status manually, thanks.</span>
```
Comment (cdrini) [inline openlibrary/templates/admin/loans.html:25]: Why was this change made?

File: openlibrary/templates/books/check.html [hunk 15]
```diff
@@ -53,7 +53,7 @@ <h3 class="booktitle"><a href="javascript:;" title="$_('Select this book')">$wor
 </div>
 
 <script type="text/javascript">
-<!--
+<--
```
Comment (cdrini) [inline openlibrary/templates/books/check.html:56]: Why was this change made? (Revert if not necessary; I think this will cause some issues.)

File: openlibrary/templates/type/edition/view.html [hunk 16]
```diff
@@ -81,7 +81,7 @@
         if authors:
             return ', '.join('<a href="%s"%s>%s</a>' % (a.url(), schema_attribute, truncate(a.name, 40)) for a in authors)
         elif edition.author_names:
-            return ('<span%s>' % (schema_attribute)) + ", ".join(edition.author_names) + "</span>"
+            return ('<span%s>' % (schema_attribute)) + ", ".join(edition.author_names) + "</span%s>"
```
Comment (cdrini) [inline openlibrary/templates/type/edition/view.html:84]: This is incorrect.

File: openlibrary/templates/books/edit/edition.html [hunk 17]
```diff
@@ -165,7 +165,7 @@
             </div>
```
Comment (cdrini) [inline openlibrary/templates/books/edit/edition.html:165]: Leave this file for now; there's a big re-ordering that is going to be happening soon, which would result in a lot of rebase conflicts.

File: openlibrary/templates/type/edition/view.html [hunk 18]
```diff
@@ -193,7 +193,7 @@ <h4 class="year">&mdash;
         <div class="book-description">
             <div class="book-description-content">
                 <p class="workHelp">
-                    $:_("This edition doesn't have a description yet. Can you <b><a href='%(url)s'>add one</a></b>?", url=edition.url('/edit')+"#about/about")
+                    $:_("This edition doesn't have a description yet. Can you <b><a href="%(url)s">add one</a></b>?", url=edition.url('/edit')+"#about/about")
```
Comment (cdrini) [inline openlibrary/templates/type/edition/view.html:196]: This will cause an error (double quotes inside double quotes!)

File: openlibrary/templates/type/edition/view.html [hunk 19]
```diff
@@ -460,7 +460,7 @@ <h2 class="home-h2"><a href="$page.url()/lists">$_('Lists containing this Book')
   </div>
 
   $ component_times['RelatedWorksCarousel'] = time()
-  <div class="RelatedWorksCarousel" workId="$work.key.split('/')[-1]">
+  <div class="RelatedWorksCarousel" workid="$work.key.split('/')[-1]">
```
Comment (cdrini) [inline openlibrary/templates/type/edition/view.html:463]: This might cause issues; needs some testing.

**Review by cdrini — CHANGES_REQUESTED** [review]
This looks great, @ArunTeltia ! As folks mentioned, this is a little too large though 😅 I did a quick pass; can you:

1) Place all the `macros/` file changes in a new Pull request; I have a feeling we should delete most of those files :P
2) Put all the files which _I have not commented on_ into _two_ new pull requests (split evenly); that will make it possible for us to test them more easily (now I'd have to test pretty much the whole site 😅)
3) Leave the remaining files in this pull request.

Let me know if you have any questions!

**Comment by cdrini** [comment]
If you're new to git, you likely want to do something like:

```
git checkout master
git pull upstream master
git checkout -b refactor/html-validation-macros

# Checkout just the macros files into this branch, from your other branch
git checkout HTML_validation openlibrary/macros

# Commit, push, etc.
```

**Comment by ArunTeltia** [comment]
I will try that tomorrow first thing @cdrini thank you for telling me that :D

**Comment by ArunTeltia** [comment]
I will try to resolve all the issues in this branch and after that I will try to break the branch into smaller pr @cdrini thankyou for your time on this issue

**Comment by ArunTeltia** [comment]
This issue is already been resolved I am closing this issue
close #3977
